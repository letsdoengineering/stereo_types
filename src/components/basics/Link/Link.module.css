.link {
    color: var(--link-blue);
}

.size-XS {
    font-size: var(--font-size-XS);
}

.size-S {
    font-size: var(--font-size-S);
}

.size-M {
    font-size: var(--font-size-M);
}

.size-L {
    font-size: var(--font-size-L);
}

.size-XL {
    font-size: var(--font-size-XL);
}

.weight-normal {
    font-weight: var(--font-weight-normal);
}

.weight-bold {
    font-weight: var(--font-weight-bold);
}

.weight-xtra {
    font-weight: var(--font-weight-xtra);
}
