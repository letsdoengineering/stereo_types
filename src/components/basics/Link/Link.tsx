import React from 'react'
import * as linkStyles from './Link.module.css'
import getClassNames from '../../../utils/get-class-names'

type LinkProps = {
    /** 16 | 18 | 20 | 30 px*/
    size?: ElementSize
    /** font weight */
    weight?: 'Normal' | 'Bold' | 'Xtra'
    /** optional: class-names */
    className?: string
    /** children always of type ReactNode */
    children?: React.ReactNode
}

/** Link: returns a customisable anchor tag component */
const Link: React.FC<LinkProps & React.AllHTMLAttributes<HTMLHtmlElement>> = ({
    size = 'S',
    weight = 'Normal',
    className,
    children,
    ...rest
}: LinkProps) => {
    const classNames = getClassNames({
        defaultClasses: [linkStyles.link, linkStyles[`size${size}`], linkStyles[`weight${weight}`]],
        className,
    })

    return (
        <a className={classNames} {...rest}>
            {children}
        </a>
    )
}

export default Link
