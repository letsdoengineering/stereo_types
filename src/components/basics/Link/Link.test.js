import * as React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { axe } from 'jest-axe'
import Link from './Link'

describe('Link', () => {
  it('has not changed the JSX layout of the page from previous snapshot', () => {
    const tree = renderer.create(<Link href=''>some link</Link>).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders the page DOM that is accessible', async () => {
    const { container } = render(<Link href=''>some link</Link>)
    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
