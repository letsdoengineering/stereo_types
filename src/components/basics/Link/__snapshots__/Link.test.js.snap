// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link has not changed the JSX layout of the page from previous snapshot 1`] = `
<a
  className="link sizeS weightNormal"
  href=""
>
  some link
</a>
`;
